apps/server/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,149 @@
+import { Controller, Get, Query, Req, Res } from "@nestjs/common";
+import * as https from "https";
+import * as dotenv from "dotenv";
+import { UsersService } from "src/users/users.service";
+dotenv.config();
+
+@Controller("auth")
+export class AuthController {
+  constructor(private usersService: UsersService) {} //Inject UserService
+
+  @Get("github")
+  githubLogin(@Req() req, @Res() res) {
+    const clientID = process.env.GITHUB_CLIENT_ID;
+    //const callbackURI = process.env.NEXT_PUBLIC_REDIRECT_URI;
+    res.redirect(
+      `https://github.com/login/oauth/authorize?client_id=${clientID}&redirect_uri=http://localhost:8080/auth/github/callback`,
+    );
+  }
+
+  @Get("github/callback")
+  async githubCallback(@Query("code") code: string, @Res() res) {
+    const clientID = process.env.GITHUB_CLIENT_ID;
+    const clientSecret = process.env.GITHUB_CLIENT_SECRET;
+    const tokenResponse = await this.exchangeCodeForToken(
+      code,
+      clientID,
+      clientSecret,
+    );
+    const githubUser = await this.fetchGithubUserProfile(
+      tokenResponse.access_token,
+    );
+
+    console.log("****GITHUB USER OBJECT******", githubUser);
```

Comment:
remove logs

Reply:
keeping this one just incase i need some of the data